### ISSUE TITLE
Unrealistic rotation behavior in Gazebo

### ISSUE THREAD

**Issue by jukindle** [jackal/jackal_simulator#18]
**Please provide the following information:**
 - OS: Ubuntu 20.04
 - ROS Distro: Noetic
 - Built from source or installed: apt-package
 - Package version: jackal_simulator 0.4.0, gazebo version 11.10.2

 **Expected behaviour**
By commanding a constant rotational velocity of 0.2rad/s, we would expect the robot to roughly rotate at 0.2rad/s about z-axis.

 **Actual behaviour**
Even though the robot rotates at the desired velocity, whenever the x-axis is axis-aligned with the diagonals in the x-y-plane, the robot seems to get stuck and the rotational velocity drops.

**To Reproduce**
Provide the steps to reproduce:
1. Install the simulator by running `sudo apt install ros-noetic-jackal-simulator`
2. Launch the simulation running `roslaunch jackal_gazebo jackal_world.launch`
3. Command the robot to rotate on spot by running

        rostopic pub -r 50 /jackal_velocity_controller/cmd_vel geometry_msgs/Twist "linear:
          x: 0.0
          y: 0.0
          z: 0.0
        angular:
          x: 0.0
          y: 0.0
          z: 0.2" 

**Other notes**
[Video of the bug](https://photos.app.goo.gl/2X3hDVr4QZMDjCQp7)

According to [this issue](https://github.com/osrf/gazebo/issues/2068) on the gazebo repository, this error seems to a bug in the friction pyramid implementation in gazebo. Two solutions were proposed: either change the friction model to cone, or increase the ERP to 0.6. Unfortunately, both solutions did not work for me and the only thing which helped is to drastically decrease the friction of the wheels. This results in unrealistic behavior as the robot would slip down slopes.

Do you see any possibility to solve this issue? I would love to use gazebo to test my software before deploying it to the real system, but with such unrealistically simulated skid-steer behavior, I am unable to do so.

**Comment by peci1** [comment]
Yeah, it would be nice if somebody found the culprit for this. I definitely don't have time for that now, but feel free to ping me for a review if you find something ;)

**Comment by peci1** [comment]
Could you maybe test it with other dynamics engines?

**Comment by AlexGisi** [comment]
The dart engine seems to be *slightly* better about this, but it is still a serious problem (Gazebo 11.11.0).
